EGE/Prog.py

```diff
@@ -271,43 +274,54 @@ def rotate_right(self):
 
 class UnOp(Op):
 
+    def __init__(self, op, arg):
+        super.__init__(op)
+        self.arg = arg
+
     def prio(self, lang): return lang.prio['`' + self.op]
 
     def to_lang_fmt(self, lang):
         return lang.get_fmt('un_op_fmt', self.op)
 
-    def children(self): return (self.arg,)
+    def children(self): return self.arg
 
 
-class Inc(UnOp)
+class Inc(UnOp):
 
     def run(self, env):
         #return eval(self.op}, '${$self->{arg}->get_ref($env)}';
         pass
 
 
 class TernaryOp(Op):
+    def __init__(self, op, arg1, arg2, arg3):
+        super.__init__(op)
+        self.arg1 = arg1
+        self.arg2 = arg2
+        self.arg3 = arg3
 
     def to_lang_fmt(self, lang):
         r = lang.get_fmt('op_fmt', self.op)
-        if not isinstance(t, list):
+        if not isinstance(r, list):
             return r
         s = make_expr(r).to_lang(lang)
         #$s =~ s/(\d+)/%$1\$s/g;
         return s
 
-    def children(self): return (self.arg1, self,arg2, self.arg3)
+    def children(self): return self.arg1, self.arg2, self.arg3
 
 
 class Var(SynElement):
+    def __init__(self, name):
+        self.name = name
 
     def to_lang(self, lang):
-        returbn lang.get_fmt('var_fmt') % self.name
+        return lang.get_fmt('var_fmt') % self.name
 
     def run(self, env):
         v = env[self.name]
```

Comment:
I'm suggesting rename variable `v` to more informative

Reply:
Tried to save logic and names from perl. I suggest to leave it for refactoring later.